sympy/printing/maple.py

```diff
@@ -0,0 +1,172 @@
+"""
+Maple code printer
+
+The MapleCodePrinter converts single sympy expressions into single
+Maple expressions, using the functions defined in the Maple objects where possible.
+
+
+FIXME: This module is still under actively developed. Some functions may be not completed.
+"""
+
+from __future__ import print_function, division
+
+from sympy.codegen.ast import Assignment
+from sympy.core import S
+from sympy.core.compatibility import string_types, range
+from sympy.printing.codeprinter import CodePrinter
+from sympy.printing.precedence import precedence, PRECEDENCE
+
+_known_func_same_name = [
+    'sin',
+    'cos',
+    'tan',
+    'sec',
+    'csc',
+    'cot',
+
+    'sinh',
+    'cosh',
+    'tanh',
+    'sech',
+    'csch',
+    'coth',
+
+    'exp',
+
+    'floor'
+
+]
+
+known_functions = {
+    # Sympy -> Maple
+    # FIXME: maybe need added
+    'Abs': 'abs',
+    'log': 'ln',
+
+    'asin': 'arcsin',
+    'acos': 'arccos',
+    'atan': 'arctan',
+    'asec': 'arcsec',
+    'acsc': 'arccsc',
+    'acot': 'arccot',
+
+    'asinh': 'arcsinh',
+    'acosh': 'arccosh',
+    'atanh': 'arctanh',
+    'asech': 'arcsech',
+    'acsch': 'arccsch',
+    'acoth': 'arccoth',
+
+    'ceiling': 'ceil',
+
+}
+
+for _func in _known_func_same_name:
+    known_functions[_func] = _func
+
+
+class MapleCodePrinter(CodePrinter):
+    """
+    Printer which converts single sympy expressions into single.
+    """
+
+    def __init__(self, settings=None):
+        if settings is None:
+            settings = dict()
+        CodePrinter.__init__(settings)
+        self.known_functions = dict(known_functions)
+        userfuncs = settings.get('user_functions', {})
+        self.known_functions.update(userfuncs)
+
+    def _get_statement(self, codestring):
+        return "%s;" % codestring
+
+    def _get_comment(self, text):
+        return "# {0}".format(text)
+
+    def _declare_number_const(self, name, value):
+        return "{0} := {1};".format(name, value.evalf(self._settings['precision']))
+
+    def _print_Pow(self, expr):
+        PREC = precedence(expr)
+        if expr.exp == -1:
+            return '1/%s' % (self.parenthesize(expr.base, PREC))
+        elif expr.exp == 0.5 or expr.exp == S(1) / 2:
+            return 'sqrt(%s)' % self._print(expr.base)
+        else:
```

Comment:
maybe add `elif ... == -1/2` here leading to `1/sqrt(`? 

And in the longer run look at an `if Rational`-type of statement with higher order roots and powers combined.

Reply:
Thanks for instruction, I agree that sqrt is more faster than x^(1/2), However, what is the advantage of a very complex root and power combination against x^(p/q)?

Reply:
Probably none. More a matter of taste I guess. Maybe I was thinking a bit too much about the visual output than the pure numerical aspects.